packages/prisma/seed-app-store.ts

```diff
@@ -157,16 +157,60 @@ async function createApp(
   keys?: Prisma.AppCreateInput["keys"],
   isTemplate?: boolean
 ) {
-  await prisma.app.upsert({
-    where: { slug },
-    create: { slug, dirName, categories, keys, enabled: true },
-    update: { dirName, categories, keys },
-  });
-  await prisma.credential.updateMany({
-    where: { type },
-    data: { appId: slug },
-  });
-  console.log(`📲 Upserted ${isTemplate ? "template" : "app"}: '${slug}'`);
+  try {
+    const foundApp = await prisma.app.findFirst({
+      /**
+       * slug and dirName both are unique and any of them can be used to find the app uniquely
+       * Using both here allows us to rename(after the app has been seeded already) `slug` or `dirName` while still finding the app to apply the change on.
+       * Note: dirName is legacy and it is same as slug for all apps created through App-Store Cli.
+       * - Take the case there was an app with slug `myvideo` and dirName `dirName-1` and that got seeded. Now, someone wants to rename the slug to `my-video`(more readable) for the app keeping dirName same.
+       *    This would make this fn to be called with slug `my-video` and dirName `dirName-1`.
+       *    Now, we can find the app because dirName would still match.
+       * - Similar, if someone changes dirName keeping slug same, we can find the app because slug would still match.
+       * - If both dirName and slug are changed, it will be added as a new entry in the DB.
+       */
+      where: {
+        OR: [
+          {
+            slug,
+          },
+          {
+            dirName,
+          },
+        ],
+      },
+    });
+    const data = { slug, dirName, categories, keys };
```

Comment:
Removed enabled:true. This is to be consistent with https://github.com/calcom/cal.com/pull/8092